### PR TITLE
Adding GetProductById and GetAllProducts endpoints

### LINKED ISSUE #9: Add two endpoints in the Product controller for the QA placement
Acceptance Criteria:

- Post and Get (Id and All) endpoints
- No unit tests must be added for the new endpoints
- No integration tests must be
- API documentation including possible responses must be added to Swagger
- API calls can be executed through Swagger UI
- Existing unit and Integration tests must execute successfully

### PR DESCRIPTION
**Pull request by Kwanele664**

### CODE REVIEW

File: We.Sell.Bread.API/Controllers/ProductController.cs [hunk 1]
```diff
@@ -1,7 +1,64 @@
...
+    {
+        _logger = logger;
+
+        _productService = new ProductService();
+    }
+
+    [HttpGet(Name = "PingProduct")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public ActionResult<string> HealthCheck()
+    {
```
Comment (Thumbeza) [inline We.Sell.Bread.API/Controllers/ProductController.cs:24]: Going the extra mile, nice

**Comment by Thumbeza** [comment]
Missing Post endpoint

File: We.Sell.Bread.API/Controllers/ProductController.cs [hunk 2]
```diff
@@ -1,7 +1,64 @@
...
+        return Ok("Product Is ALive");
+    }
+
```
Comment (Thumbeza) [inline We.Sell.Bread.API/Controllers/ProductController.cs:27]: Missing PostAsync endpoint

Comment (Kwanele664) [reply We.Sell.Bread.API/Controllers/ProductController.cs:27]: added PostAsync endpoint for product

File: We.Sell.Bread.Infrastructure/DataFiles/Product.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+[
+  {
+    "ProductId": "411fbad3-925e-4044-9269-f962641f9277",
+    "ProductName": "Pumpernickel Bread",
+    "Price": 20.00,
+    "Description": "A dense, dark bread made from coarsely ground whole rye grains",
+    "StockQuantity": 7
+  },
+  {
+    "ProductName": "Multigrain Bread",
```
Comment (Thumbeza) [inline We.Sell.Bread.Infrastructure/DataFiles/Product.json:11]: this product has no id, either delete it from the file or add id

Comment (Kwanele664) [reply We.Sell.Bread.Infrastructure/DataFiles/Product.json:11]: I have added the missing productId

**Review by Thumbeza — APPROVED** [review]

**Pull request merged**